java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/MainUtilities.java

```diff
@@ -213,6 +208,33 @@ public static void main(String[] args) throws Exception {
 		}
 	}
 
+	private static void generateObfFromGpx(List<String> subArgs) throws IOException, SQLException,
+			XmlPullParserException, InterruptedException {
+		if (subArgs.size() != 0) {
+			File file = new File(subArgs.get(0));
+			if (file.isDirectory() || file.getName().endsWith(GPX_FILE_EXT) || file.getName().endsWith(".gpx.gz")) {
+				OsmGpxWriteContext.QueryParams qp = new OsmGpxWriteContext.QueryParams();
+				qp.osmFile = File.createTempFile(Algorithms.getFileNameWithoutExtension(file), ".osm");
+				OsmGpxWriteContext ctx = new OsmGpxWriteContext(qp);
+				File tmpFolder = new File(file.getParentFile(), String.valueOf(System.currentTimeMillis()));
+				String path = file.isDirectory() ? file.getAbsolutePath() : file.getParentFile().getPath();
+				File targetObf = new File(path, Algorithms.getFileNameWithoutExtension(file) + BINARY_MAP_INDEX_EXT);
+				List<File> files = new ArrayList<>();
+				if (file.isDirectory()) {
+					files = Arrays.asList(Objects.requireNonNull(file.listFiles()));
+				} else {
+					files.add(file);
+				}
+				if (!files.isEmpty()) {
+					ctx.writeObf(files, tmpFolder, Algorithms.getFileNameWithoutExtension(file), targetObf);
+				}
+				qp.osmFile.deleteOnExit();
+			}
+		} else {
+			System.out.println("Usage: <path_to_directory_with_gpx_files> or <path_to_gpx_file_with_file_name>");
```

Comment:
This is not visible in main help of Main utilties - check what's print in main utilities seciton help